### ISSUE TITLE
Move `isStrict` method to base WebSocket message class

### ISSUE THREAD

**Issue by harpocrates** [akka/akka-http#3844]
### Short description

As far as I can tell, all WebSocket message types have a `def isStrict(): Boolean` method. I propose to move this method to the base class `akka.http.javadsl.model.ws.Message`.

### Details

Since the `def isStrict(): Boolean` is defined on `akka.http.javadsl.model.ws.BinaryMessage` and `akka.http.javadsl.model.ws.TextMessage` but not on `akka.http.javadsl.model.ws.Message`, I have to do an unnecessary pattern match in order to call `.isStrict` on a `akka.http.javadsl.model.ws.Message` (or a `akka.http.scaladsl.model.ws.Message`):

```scala
val isStrict: Boolean = message match {
  case b: BinaryMessage => b.isStrict
  case t: TextMessage => t.isStrict
}
```

It would be more ergonomic to be able to just do

```scala
val isStrict: Boolean = message.isStrict
```

**Comment by johanandren** [comment]
I think you pretty much always will have to determine binary or text anyway, but I agree that it makes sense to have `isStrict` on `Message`. Would you be up for a PR adding it?

**Comment by harpocrates** [comment]
> I think you pretty much always will have to determine binary or text anyway, but I agree that it makes sense to have isStrict on Message. Would you be up for a PR adding it?

I wanted `isStrict` in the context of detecting and debugging some entity subscription timeouts for non-strict payloads. The motivation is similar to `isText: Boolean`.

**Comment by jrudolph** [comment]
Fixed in #3847.
